src/PointwiseFunctions/AnalyticData/GrMhd/MagnetizedTovStar.hpp

```diff
@@ -0,0 +1,234 @@
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"
+#include "PointwiseFunctions/Hydro/TagsDeclarations.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace grmhd::AnalyticData {
+/*!
+ * \brief Magnetized TOV star initial data, where the Einstein constraint
+ * equations are not solved for the metric.
+ *
+ * Superposes a poloidal magnetic field on top of a TOV solution where the
+ * vector potential has the form
+ *
+ * \f{align*}{
+ *  A_{\phi} = A_b \rho^2 \max(p-p_{\mathrm{cut}}, 0)^{n_s}
+ * \f}
+ *
+ * where \f$A_b\f$ controls the amplitude of the magnetic field,
+ * \f$\rho^2=x^2+y^2=r^2-z^2\f$ is the cylindrical radius,
+ * \f$n_s\f$ controls the degree if differentiability, and
+ * \f$p_{\mathrm{cut}}\f$ controls the pressure cutoff below which the magnetic
+ * field is zero.
+ *
+ * In Cartesian coordinates the vector potential is:
+ *
+ * \f{align*}{
+ *   A_x&=-\frac{y}{\varpi^2}A_\phi, \\
+ *   A_y&=\frac{x}{\varpi^2}A_\phi, \\
+ *   A_z&=0,
+ * \f}
+ *
+ * For the poloidal field this means
+ *
+ * \f{align*}{
+ *   A_x &= -y A_b\max(p-p_{\mathrm{cut}}, 0)^{n_s}, \\
+ *   A_y &= x A_b\max(p-p_{\mathrm{cut}}, 0)^{n_s}.
+ * \f}
+ *
+ * The magnetic field is computed from the vector potential as
+ *
+ * \f{align*}{
+ *   B^i&=n_a\epsilon^{aijk}\partial_jA_k \\
+ *      &=-\frac{1}{\sqrt{\gamma}}[ijk]\partial_j A_k,
+ * \f}
+ *
+ * where \f$[ijk]\f$ is the total anti-symmetric symbol. This means that
+ *
+ * \f{align*}{
+ *   B^x&=\frac{1}{\sqrt{\gamma}} (\partial_z A_y-\partial_y A_z), \\
+ *   B^y&=\frac{1}{\sqrt{\gamma}} (\partial_x A_z - \partial_z A_x), \\
+ *   B^z&=\frac{1}{\sqrt{\gamma}} (\partial_y A_x - \partial_x A_y).
+ * \f}
+ *
+ * Focusing on the region where the field is non-zero we have:
+ *
+ * \f{align*}{
+ *   \partial_x A_y
+ *   &= A_b(p-p_{\mathrm{cut}})^{n_s}+
+ *     \frac{x^2}{r}A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_r p \\
+ *   \partial_y A_x
+ *   &= -A_b(p-p_{\mathrm{cut}})^{n_s}-
+ *     \frac{y^2}{r}A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_r p \\
+ *   \partial_z A_x
+ *   &= -\frac{yz}{r}A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_rp \\
+ *   \partial_z A_y
+ *   &= \frac{xz}{r}A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_rp
+ * \f}
+ *
+ * The magnetic field is given by:
+ *
+ * \f{align*}{
+ *   B^x&=\frac{1}{\sqrt{\gamma}}\frac{xz}{r}
+ *        A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_rp \\
+ *   B^y&=\frac{1}{\sqrt{\gamma}}\frac{yz}{r}
+ *        A_bn_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_rp \\
+ *   B^z&=-\frac{A_b}{\sqrt{\gamma}}\left[
+ *        2(p-p_{\mathrm{cut}})^{n_s} \phantom{\frac{a}{b}}\right. \\
+ *      &\left.+\frac{x^2+y^2}{r}
+ *        n_s(p-p_{\mathrm{cut}})^{n_s-1}\partial_r p
+ *        \right]
+ * \f}
+ *
+ * Please note that at \f$r=0\f$ the magnetic field is
```

Comment:
"Please" in documentation feels funny. How about "Taking the small-\f$r\f$ limit gives the \f$r=0\f$ magnetic field:"